pkg/oauthserver/osinserver/registrystorage/storage.go

```diff
@@ -290,28 +290,48 @@ func (s *storage) convertFromAccessToken(access *oauthapi.OAuthAccessToken) (*os
 	}, nil
 }
 
-func convertFromUser(user interface{}) (name, uid string, err error) {
+func convertFromUser(user interface{}) (name, uid, idpName string, idpGroups []string, err error) {
 	info, ok := user.(kuser.Info)
 	if !ok {
-		return "", "", fmt.Errorf("did not receive user.Info: %#v", user) // should be impossible
+		return "", "", "", nil, fmt.Errorf("did not receive user.Info: %#v", user) // should be impossible
 	}
 
 	name = info.GetName()
 	uid = info.GetUID()
 	if len(name) == 0 || len(uid) == 0 {
-		return "", "", fmt.Errorf("user.Info has no user name or UID: %#v", info) // should be impossible
+		return "", "", "", nil, fmt.Errorf("user.Info has no user name or UID: %#v", info) // should be impossible
 	}
 
-	return name, uid, nil
+	if userIdentityMetadata, ok := user.(api.UserIdentityMetadata); ok {
+		idpName = userIdentityMetadata.GetIdentityProviderName()
+		idpGroups = userIdentityMetadata.GetIdentityProviderGroups()
+		if len(idpName) == 0 || len(idpGroups) == 0 {
```

Comment:
Why shouldn't it be possible to have 0 groups ?

Reply:
It is possible to have zero groups.  However, it is not possible to say "I have groups" (which is what `UserIdentityMetadata` signals) and then not supply any groups - that is a programmer error.

User and UID are required via the `user.Info`.  Provider name and groups are only required if you actually implement `UserIdentityMetadata`, which only happens when there are groups - see `pkg/oauthserver/userregistry/identitymapper/group_mapper.go`.  Basically I am just enforcing some sanity on the `interface{}` that is held in the `UserData` field.